src/pages/iou/request/step/IOURequestStepParticipants.js

```diff
@@ -100,6 +109,22 @@ function IOURequestStepParticipants({
             }
 
             IOU.setMoneyRequestParticipants_temporaryForRefactor(transactionID, val);
+
+            // change customUnitRateID when choosing
+            if (val[0].isPolicyExpenseChat && TransactionUtils.isCustomUnitRateIDForP2P(transaction)) {
+                let customUnitRateID = '';
+                if (val[0].policyID && lastSelectedDistanceRates[val[0].policyID]) {
+                    customUnitRateID = lastSelectedDistanceRates[val[0].policyID];
+                } else {
+                    const policy = ReportUtils.getPolicy(val[0].policyID);
```

Comment:
This function is marked deprecated. Can we pass the policy coming from withOnyx instead?

Reply:
how should I do this here as I don't have policy id until I choose a participant?

Reply:
Oh yeah good point. We could connect withOnyx to the transactionDraft.participants, use a selector to grab the policy from the first, participants, and then run some logic whenever that updates.

However, that feels worse, because I don't think we like to have an action such as `setMoneyRequestParticipants_temporaryForRefactor` trigger another action `updateDistanceRequestRate`. I'm good with what you have here. If you're curious, you can ask for advice about our guidelines on this in #expensify-open-source, but I think it's fine.